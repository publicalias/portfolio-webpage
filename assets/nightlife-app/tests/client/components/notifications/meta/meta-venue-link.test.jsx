"use strict";

//local imports

const MetaVenueLink = require("../../../../../scripts/client/components/notifications/meta/meta-venue-link");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(MetaVenueLink);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(MetaVenueLink));

//meta venue link

describe("MetaVenueLink", () => {

  const testSnapshot = withDataList(initTestSnapshot(testShallow), [null, { venue: { id: "id-a" } }]);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { venue: { name: "Venue A" } }));

});
