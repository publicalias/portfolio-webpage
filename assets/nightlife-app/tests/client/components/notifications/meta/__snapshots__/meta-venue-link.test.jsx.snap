// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaVenueLink should match snapshot (default) 1`] = `
<Link
  to="/venues/page/id-a"
>
  <span
    className="u-underline"
  >
    Undefined
  </span>
</Link>
`;

exports[`MetaVenueLink should match snapshot (name) 1`] = `
<Link
  to="/venues/page/id-a"
>
  <span
    className="u-underline"
  >
    Venue A
  </span>
</Link>
`;
